lib/sycamore/sycamore/utils/pdf.py

```diff
@@ -108,14 +113,45 @@ def read_stderr(fh, q):
         t_err.join()
 
 
-def convert_from_path_streamed_batched(filename: str, batch_size: int) -> Generator[List[Image.Image], None, None]:
+def convert_from_path_streamed_batched(
+    filename: str, batch_size: int, hash_key: str, use_cache: bool = True
+) -> Generator[List[Image.Image], None, None]:
     """Note: model service will call this to get batches of images for processing"""
+    images = pdf_to_ppm_cache.get(hash_key) if use_cache else None
+    if images:
+        print(f"PDFToPPM Cache Hit. Getting the results from cache.")
+        yield from yield_batches(images, batch_size)
+    else:
+        print(f"PDFToPPM Cache Miss. Getting the results.")
+        yield from load_images(filename, batch_size, use_cache, hash_key)
+
+
+def yield_batches(images: list, batch_size: int) -> Generator[List[Image.Image], None, None]:
+    batch = []
+    for image in images:
+        batch.append(image)
+        if len(batch) == batch_size:
+            yield batch
+            batch = []
+    if len(batch) > 0:
+        yield batch
+
+
+def load_images(
+    filename: str, batch_size: int, use_cache: bool, hash_key: str
+) -> Generator[List[Image.Image], None, None]:
     batch = []
+    all_images = []
     for i in convert_from_path_streamed(filename):
         batch.append(i)
         if len(batch) == batch_size:
             yield batch
+            all_images.extend(batch)
             batch = []
 
     if len(batch) > 0:
+        all_images.extend(batch)
         yield batch
```

Comment:
Is there a difference between calling all_images.extend() before or after yield.  You have them in different orders.

Reply:
it's the same